### DISCUSSION TITLE
Sharpen and also darken the text

### DISCUSSION THREAD

**Discussion by srini8080** [ImageMagick/ImageMagick#5217] [Digital Image Processing]
Version: ImageMagick 7.1.0-35 Q16-HDRI x86_64 20062 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5) 
Delegates (built-in): bzlib djvu fftw fontconfig freetype heic jbig jng jp2 jpeg lzma pangocairo png tiff webp x xml zlib
Compiler: gcc (8.5)

This text is blur when zoomed. Is there a way to make it look darker and sharper?

![00004-done](https://user-images.githubusercontent.com/12046302/173231035-a864afdf-210c-4c65-b76f-a84dcde49f06.png)

**Comment by fmw42** [comment]
What is your Imagemagick version and platform/OS?  What was your Imagemagick command to "zoom"?  Raster text is always going to be somewhat blurry when resized larger. You can try to use some -unsharp after resizing.

**Reply by srini8080** [reply]
Centos 8.3

Version: ImageMagick 7.1.0-35 Q16-HDRI x86_64 20062 https://imagemagick.org/
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5)
Delegates (built-in): bzlib djvu fftw fontconfig freetype heic jbig jng jp2 jpeg lzma pangocairo png tiff webp x xml zlib
Compiler: gcc (8.5)

Zoom is just in the image viewer.

I tried unsharp 0x3.0. Did not work well. Is there anything that can be done to sharpen the text? Morphology? or anything else?

Thanks,

**Reply by fmw42** [reply]
Nothing that I know other than perhaps AI/Deep Learning for super-resolution.  But Imagemagick does not have that feature.

**Comment by snibgo** [comment]
@srini8080: The image you show is strange. It seems to be made of dots, each with a diameter of about 6 pixels. There is no detail at a finer scale.

I suppose that image has been processed somehow. I suggest that instead, you should start from an unprocessed image.

**Reply by srini8080** [reply]
![00004](https://user-images.githubusercontent.com/12046302/173270945-916466ed-e296-4e14-a0e9-0e5872b48cb7.jpg)

Here is the original. Wanted to make the background white (remove the sun burn). Make the text darker and sharper.

Thanks,

**Reply by fmw42** [reply]
Try 

```
convert 173270945-916466ed-e296-4e14-a0e9-0e5872b48cb7.jpg \
-negate -lat 20x20+5% -negate \
result.png

```

![result](https://user-images.githubusercontent.com/13458218/173276880-fde81584-e7d9-454f-91ef-589de8561603.png)


Adjust the 5% lower for darker or higher for lighter.

**Reply by fmw42** [reply]
By the way, you asked a similar question at https://github.com/ImageMagick/ImageMagick/discussions/5192. Did you try that processing?

**Reply by srini8080** [reply]
I tried  convert y.jpg -resize 200%x200% -threshold 80% -sharpen 0x3.0 x.jpg and the letters and the image lines are getting thickened but not sharp.

**Reply by fmw42** [reply]
Apart from using -unsharp or -sharpen, I do not know anything Imagemagick can do to sharpen your letters when enlarging.  Only AI would be able to do that effectively. However, perhaps you should try -magnify with one of the 2x art file enlarging methods. That is the only thing that I can suggest.  See https://imagemagick.org/script/command-line-options.php#magnify
